inspire_disambiguation/api.py

```diff
@@ -83,27 +86,40 @@ def train_and_save_distance_model(
     LOGGER.info("Pulling training data from ES")
     start_time = datetime.now()
     curated_signatures = get_signatures(only_curated=True)
-    input_clusters = get_input_clusters(curated_signatures)
+    # splitting signatures
+    train_signatures = random.sample(curated_signatures, k=int(len(curated_signatures) * 0.8))
+    test_signatures = [signature for signature in curated_signatures if
```

Comment:
this is very expensive (quadratic) for long lists. You probably want to use `set`s here (which have `O(1)` membership test instead of `O(N)`.

Reply:
But set can't be made from Signature object list (they're not hashable), so still I'll have to iterate through it to get signature_uuids

Reply:
then you need to be smarter, and use something like a `dict` where the key is the `signature_uuid`.